ouroboros-network/src/Ouroboros/Network/Protocol/ChainSync/Codec.hs

```diff
@@ -109,6 +115,24 @@ codecChainSync encodeHeader decodeHeader encodePoint decodePoint =
 
         _ -> fail ("codecChainSync: unexpected key " ++ show (key, len))
 
+    encodeHeaderWrapped :: header -> CBOR.Encoding
+    encodeHeaderWrapped header =
+        CBOR.encodeTag 24
+     <> CBOR.encodeBytes (CBOR.toStrictByteString (encodeHeader header))
+
+    decodeHeaderWrapped :: forall s. CBOR.Decoder s header
+    decodeHeaderWrapped = do
+      tag <- CBOR.decodeTag
+      when (tag /= 24) $ fail "expected tag 24 (CBOR-in-CBOR)"
+      payload <- CBOR.decodeBytes
+      case CBOR.deserialiseFromBytes (decodeHeader payload)
+                                     (LBS.fromStrict payload) of
+        Left (CBOR.DeserialiseFailure _ reason) -> fail reason
+        Right (trailing, header)
+          | not (LBS.null trailing) -> fail "trailing bytes in CBOR-in-CBOR"
+          | otherwise               -> return header
+
```

Comment:
Consider using `Cardano.Binary.encodeUnknownCborDataItem`/`Cardano.Binary.decodeUnknownCborDataItem`?

Reply:
We don't depend on that package at the moment. And if we did, I'd want to rename those things. In the context we're using them they're not unknown. The unknown thing is a specific use of CBOR-in-CBOR.